src/main/java/luca/parser/Parser.java

```diff
@@ -0,0 +1,300 @@
+            throw new InvalidArgumentException(":-( OOPS!!! Start-End date/time of event "
+                    + "cannot be empty.");
+        }
+        return new CreateEventCommand(description, start, end);
+    }
+
+    /**
+     * Parse input tokens and creates LocalDateTime object.
+     *
+     * @param stringDate date as a string.
+     * @param stringTime time as a string.
+     * @return LocalDateTime object
+     * @throws NumberFormatException if integer is not given.
+     * @throws DateTimeException if the date format is incorrect.
+     */
+    private static LocalDateTime parseDateTime(String stringDate, String stringTime)
+            throws NumberFormatException, DateTimeException {
+        int time;
+
+        time = Integer.parseInt(stringTime);
+
+        String hours;
+        String minutes;
+
+        if (time / 100 > 9) {
+            hours = Integer.toString(time / 100);
+        } else {
+            hours = "0" + (time / 100);
+        }
+        if (time % 100 > 9) {
+            minutes = Integer.toString(time % 100);
+        } else {
+            minutes = "0" + (time % 100);
+        }
+        return LocalDateTime.parse(stringDate + "T" + hours + ":" + minutes + ":00");
+    }
+
+    /**
+     * Parse input tokens and creates a FindCommand.
+     *
+     * @param tokens input tokens to be parsed.
+     * @return FindCommand.
+     * @throws InvalidArgumentException if there are invalid arguments.
+     */
+    private static Command parseFindCommand(String[] tokens) throws InvalidArgumentException {
+        if (tokens.length != 2) {
+            throw new InvalidArgumentException(":-( OOPS!!! Please enter only one keyword "
+                    + "following the find command");
+        }
+
+        return new FindCommand(tokens[1]);
+    }
+
+    /**
+     * Parse through the command and creates a command accordingly.
+     *
+     * @param input user input string.
+     * @return Command to be run.
+     * @throws DukeException thrown when incorrect format of invalid arguments are used.
+     */
+    public static Command parse(String input) throws DukeException {
+        String[] tokens = input.split("\\s+");
+        Command command;
+        switch (tokens[0]) {
+        case "list":
+            command = new ListCommand();
+            break;
+        case "mark":
+            command = parseMarkCommand(tokens);
+            break;
+        case "unmark":
+            command = parseUnmarkCommand(tokens);
+            break;
+        case "todo":
+            command = parseTodoCommand(tokens);
+            break;
+        case "deadline":
+            command = parseDeadlineCommand(tokens);
+            break;
+        case "event":
+            command = parseEventCommand(tokens);
+            break;
+        case "delete":
+            command = parseDeleteCommand(tokens);
+            break;
+        case "bye":
+            command = new ExitCommand();
+            break;
```

Comment:
If user entered **"bye do somthing /at Sunday"**, your program will end directly. Maybe it can be recognized as an "invalid user input (user typo)" or to double confirm whether user wants to quit or not.